internal/host/transition.go

```diff
@@ -409,6 +410,18 @@ func (th *transitionHandler) HasClusterError(sw stateswitch.StateSwitch, args st
 	return swag.StringValue(cluster.Status) == models.ClusterStatusError, nil
 }
 
+func (th *transitionHandler) HasInstallationTimedOut(sw stateswitch.StateSwitch, args stateswitch.TransitionArgs) (bool, error) {
+	sHost, ok := sw.(*stateHost)
+	if !ok {
+		return false, errors.New("HasInstallationTimedOut incompatible type of StateSwitch")
+	}
+	maxDuration, ok := InstallationProgressTimeout[sHost.host.Progress.CurrentStage]
```

Comment:
There are more stages that are not in this map. 
These are just the ones we have timeout for, please handle the case where the host current stage isn't in the map (return false).

Reply:
@eranco74 default 1 hour?